Looks like syncing tt_llk to latest breaks kernels.  When building for BH I got the following error:
```
           BuildKernels | ERROR    | trisc1 compile failure -- cmd: cd /home/ruizhang/.cache/tt-metal-cache/fe5c7237af/1052672/kernels/void_compute_kernel/10117558232137305589/trisc1/ && /proj_sw/user_dev/ruizhang/tt-metal/runtime/sfpi/compiler/bin/riscv32-tt-elf-g++ -O3 -mcpu=tt-bh -fno-rvtt-sfpu-replay -std=c++17 -flto -ffast-math -fno-use-cxa-atexit -fno-exceptions -Wall -Werror -Wno-unknown-pragmas -Wno-deprecated-declarations -Wno-error=multistatement-macros -Wno-error=parentheses -Wno-error=unused-but-set-variable -Wno-unused-variable -Wno-unused-function -I. -I.. -I/proj_sw/user_dev/ruizhang/tt-metal/ -I/proj_sw/user_dev/ruizhang/tt-metal/ttnn -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/include -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/inc -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hostdevcommon/api -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/inc/debug -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/inc/blackhole -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/inc/blackhole/blackhole_defines -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/inc/blackhole/noc -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/common -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_io -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/third_party/tt_llk/tt_llk_blackhole/common/inc -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/api/ -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/api/tt-metalium/ -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/third_party/tt_llk/tt_llk_blackhole/llk_lib -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/common -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_io -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu -I/proj_sw/user_dev/ruizhang/tt-metal/runtime/sfpi/include -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/firmware/src -I/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/third_party/tt_llk/tt_llk_blackhole/llk_lib -c -o trisck.o /proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/firmware/src/trisck.cc -DARCH_BLACKHOLE -DIS_NOT_POW2_NUM_L1_BANKS=1 -DLOG_BASE_2_OF_NUM_DRAM_BANKS=3 -DNUM_DRAM_BANKS=8 -DNUM_L1_BANKS=130 -DPCIE_NOC_X=19 -DPCIE_NOC_Y=24 -DTENSIX_FIRMWARE -DLOCAL_MEM_EN=0 -DUCK_CHLKC_MATH -DNAMESPACE=chlkc_math -DCOMPILE_FOR_TRISC=1 -DKERNEL_BUILD -DDISPATCH_MESSAGE_ADDR=4290184248 
terminate called after throwing an instance of 'std::runtime_error'
  what():  TT_THROW @ /proj_sw/user_dev/ruizhang/tt-metal/tt_metal/impl/program/program.cpp:132: tt::exception
info:
Failed to generate binaries for void_compute_kernel TT_THROW @ /proj_sw/user_dev/ruizhang/tt-metal/tt_metal/jit_build/build.cpp:66: tt::exception
info:
trisc1 build failed. Log: In file included from /proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/llk_math_eltwise_unary_sfpu_trigonometry.h:9,
                 from /proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_math_unary_sfpu_api.h:28,
                 from /proj_sw/user_dev/ruizhang/tt-metal/tt_metal/include/compute_kernel_api.h:22,
                 from /proj_sw/user_dev/ruizhang/tt-metal/tt_metal/programming_examples/hello_world_compute_kernel/kernels/compute/void_compute_kernel.cpp:6,
                 from ../chlkc_math.cpp:3,
                 from /proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/common/chlkc_list.h:21,
                 from /proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/firmware/src/trisck.cc:14:
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_trigonometry.h: In function 'void ckernel::sfpu::calculate_tangent()':
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_trigonometry.h:84:33: error: reference to 'FRAC_1_PI' is ambiguous
   84 |         vFloat v = dst_reg[0] * FRAC_1_PI;
      |                                 ^~~~~~~~~
In file included from /proj_sw/user_dev/ruizhang/tt-metal/tt_metal/third_party/tt_llk/tt_llk_blackhole/common/inc/ckernel_sfpu.h:45,
                 from /proj_sw/user_dev/ruizhang/tt-metal/tt_metal/third_party/tt_llk/tt_llk_blackhole/common/inc/cmath_common.h:12,
                 from /proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_math_common_api.h:10,
                 from /proj_sw/user_dev/ruizhang/tt-metal/tt_metal/include/compute_kernel_api.h:18:
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/third_party/tt_llk/tt_llk_blackhole/common/inc/sfpu/ckernel_sfpu_trigonometry.h:16:18: note: candidates are: 'constexpr const double ckernel::sfpu::{anonymous}::FRAC_1_PI'
   16 | constexpr double FRAC_1_PI = 0.3183098861837907;
      |                  ^~~~~~~~~
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_trigonometry.h:21:20: note:                 'const float ckernel::sfpu::FRAC_1_PI'
   21 | static const float FRAC_1_PI = 0.31830987f;
      |                    ^~~~~~~~~
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_trigonometry.h: At global scope:
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_trigonometry.h:95:20: error: ambiguous template specialization 'sfpu_sinpi<true>' for 'sfpi::vFloat ckernel::sfpu::sfpu_sinpi(sfpi::vFloat)'
   95 | sfpi_inline vFloat sfpu_sinpi<true>(vFloat x) {
      |                    ^~~~~~~~~~~~~~~~
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_trigonometry.h:92:15: note: candidates are: 'template<bool APPROXIMATION_MODE> sfpi::vFloat ckernel::sfpu::sfpu_sinpi(sfpi::vFloat)'
   92 | static vFloat sfpu_sinpi(vFloat x);
      |               ^~~~~~~~~~
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/third_party/tt_llk/tt_llk_blackhole/common/inc/sfpu/ckernel_sfpu_trigonometry.h:19:21: note:                 'template<bool APPROXIMATION_MODE> sfpi::vFloat ckernel::sfpu::{anonymous}::sfpu_sinpi(sfpi::vFloat)'
   19 | static sfpi::vFloat sfpu_sinpi(sfpi::vFloat x);
      |                     ^~~~~~~~~~
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_trigonometry.h:102:20: error: ambiguous template specialization 'sfpu_sinpi<false>' for 'sfpi::vFloat ckernel::sfpu::sfpu_sinpi(sfpi::vFloat)'
  102 | sfpi_inline vFloat sfpu_sinpi<false>(vFloat x) {
      |                    ^~~~~~~~~~~~~~~~~
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_trigonometry.h:92:15: note: candidates are: 'template<bool APPROXIMATION_MODE> sfpi::vFloat ckernel::sfpu::sfpu_sinpi(sfpi::vFloat)'
   92 | static vFloat sfpu_sinpi(vFloat x);
      |               ^~~~~~~~~~
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/third_party/tt_llk/tt_llk_blackhole/common/inc/sfpu/ckernel_sfpu_trigonometry.h:19:21: note:                 'template<bool APPROXIMATION_MODE> sfpi::vFloat ckernel::sfpu::{anonymous}::sfpu_sinpi(sfpi::vFloat)'
   19 | static sfpi::vFloat sfpu_sinpi(sfpi::vFloat x);
      |                     ^~~~~~~~~~
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_trigonometry.h: In function 'void ckernel::sfpu::calculate_sine()':
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_trigonometry.h:113:33: error: reference to 'FRAC_1_PI' is ambiguous
  113 |         vFloat v = dst_reg[0] * FRAC_1_PI;
      |                                 ^~~~~~~~~
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/third_party/tt_llk/tt_llk_blackhole/common/inc/sfpu/ckernel_sfpu_trigonometry.h:16:18: note: candidates are: 'constexpr const double ckernel::sfpu::{anonymous}::FRAC_1_PI'
   16 | constexpr double FRAC_1_PI = 0.3183098861837907;
      |                  ^~~~~~~~~
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_trigonometry.h:21:20: note:                 'const float ckernel::sfpu::FRAC_1_PI'
   21 | static const float FRAC_1_PI = 0.31830987f;
      |                    ^~~~~~~~~
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_trigonometry.h: In function 'void ckernel::sfpu::calculate_cosine()':
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_trigonometry.h:129:33: error: reference to 'FRAC_1_PI' is ambiguous
  129 |         vFloat v = dst_reg[0] * FRAC_1_PI + 0.5f;
      |                                 ^~~~~~~~~
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/third_party/tt_llk/tt_llk_blackhole/common/inc/sfpu/ckernel_sfpu_trigonometry.h:16:18: note: candidates are: 'constexpr const double ckernel::sfpu::{anonymous}::FRAC_1_PI'
   16 | constexpr double FRAC_1_PI = 0.3183098861837907;
      |                  ^~~~~~~~~
/proj_sw/user_dev/ruizhang/tt-metal/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_trigonometry.h:21:20: note:                 'const float ckernel::sfpu::FRAC_1_PI'
   21 | static const float FRAC_1_PI = 0.31830987f;
      |                    ^~~~~~~~~

backtrace:
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x212088) [0x754ce3b66088]
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x39a9aa) [0x754ce3cee9aa]
 --- tt::tt_metal::JitBuildState::compile_one(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, tt::tt_metal::JitBuildSettings const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x3a0868) [0x754ce3cf4868]
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x3a07e2) [0x754ce3cf47e2]
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x30fb6f) [0x754ce3c63b6f]
 --- /lib/x86_64-linux-gnu/libc.so.6(+0x99ee8) [0x754ce2d57ee8]
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x381349) [0x754ce3cd5349]
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x3a05ee) [0x754ce3cf45ee]
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x3a43c5) [0x754ce3cf83c5]
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x3a2ffd) [0x754ce3cf6ffd]
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x3a23f8) [0x754ce3cf63f8]
 --- /lib/x86_64-linux-gnu/libstdc++.so.6(+0xdc253) [0x754ce2fe5253]
 --- /lib/x86_64-linux-gnu/libc.so.6(+0x94ac3) [0x754ce2d52ac3]
 --- /lib/x86_64-linux-gnu/libc.so.6(+0x126850) [0x754ce2de4850]

backtrace:
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x24cf78) [0x754ce3ba0f78]
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x240214) [0x754ce3b94214]
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x3a0868) [0x754ce3cf4868]
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x3a07e2) [0x754ce3cf47e2]
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x30fb6f) [0x754ce3c63b6f]
 --- /lib/x86_64-linux-gnu/libc.so.6(+0x99ee8) [0x754ce2d57ee8]
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x381349) [0x754ce3cd5349]
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x3a05ee) [0x754ce3cf45ee]
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x3a43c5) [0x754ce3cf83c5]
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x3a2ffd) [0x754ce3cf6ffd]
 --- /proj_sw/user_dev/ruizhang/tt-metal/build_Release_tracy/tt_metal/libtt_metal.so(+0x3a23f8) [0x754ce3cf63f8]
 --- /lib/x86_64-linux-gnu/libstdc++.so.6(+0xdc253) [0x754ce2fe5253]
 --- /lib/x86_64-linux-gnu/libc.so.6(+0x94ac3) [0x754ce2d52ac3]
 --- /lib/x86_64-linux-gnu/libc.so.6(+0x126850) [0x754ce2de4850]

Aborted (core dumped)
```

--- a/tt_metal/hw/inc/firmware_common.h
+++ b/tt_metal/hw/inc/firmware_common.h
@@ -1,4 +1,4 @@
-// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+// SPDX-FileCopyrightText: © 2023 Tenstorrent AI ULC
 //
 // SPDX-License-Identifier: Apache-2.0
 

--- a/tt_metal/hw/inc/firmware_common.h
+++ b/tt_metal/hw/inc/firmware_common.h
@@ -131,8 +131,8 @@ bool is_message_go() {
 #define EARLY_RETURN_FOR_DEBUG
 #endif
 
-inline __attribute__((always_inline)) void disable_gathering() {
-#if defined(ARCH_BLACKHOLE)
+inline __attribute__((always_inline)) void configure_gathering() {
+#if defined(ARCH_BLACKHOLE) && !defined(ENABLE_GATHERING)
     // Workaround for tt-metal#16439, making sure gathering multiple instructions issued to Tensix is disabled
     // Brisc does not issue Tensix instructions but to be consistent for all riscs around Tensix we disable it
     // Disable gathering: set bit 18
@@ -190,7 +190,7 @@ inline __attribute__((always_inline)) void disable_relaxed_memory_ordering() {
 }
 
 inline __attribute__((always_inline)) void configure_csr() {
-    disable_gathering();
+    configure_gathering();
     configure_l1_data_cache();
     disable_relaxed_memory_ordering();
 }

--- a/tt_metal/jit_build/build.cpp
+++ b/tt_metal/jit_build/build.cpp
@@ -242,6 +242,10 @@ void JitBuildEnv::init(
         this->defines_ += "-DDISABLE_RELAXED_MEMORY_ORDERING ";
     }
 
+    if (rtoptions.get_gathering_enabled()) {
+        this->defines_ += "-DENABLE_GATHERING ";
+    }
+
     if (tt::tt_metal::MetalContext::instance().get_cluster().is_base_routing_fw_enabled()) {
         this->defines_ += "-DROUTING_FW_ENABLED ";
     }

--- a/tt_metal/jit_build/build.cpp
+++ b/tt_metal/jit_build/build.cpp
@@ -1,4 +1,4 @@
-// SPDX-FileCopyrightText: © 2025 Tenstorrent Inc.
+// SPDX-FileCopyrightText: © 2023 Tenstorrent AI ULC
 //
 // SPDX-License-Identifier: Apache-2.0
 

--- a/tt_metal/llrt/rtoptions.cpp
+++ b/tt_metal/llrt/rtoptions.cpp
@@ -1,4 +1,4 @@
-// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+// SPDX-FileCopyrightText: © 2023 Tenstorrent AI ULC
 //
 // SPDX-License-Identifier: Apache-2.0
 

--- a/tt_metal/llrt/rtoptions.cpp
+++ b/tt_metal/llrt/rtoptions.cpp
@@ -8,7 +8,6 @@
 #include <stdio.h>
 #include <cstdlib>
 #include <cstring>
-#include <functional>
 #include <stdexcept>
 #include <string>
 
@@ -190,6 +189,10 @@ RunTimeOptions::RunTimeOptions() {
 
     if (getenv("TT_METAL_DISABLE_RELAXED_MEM_ORDERING")) {
         this->disable_relaxed_memory_ordering = true;
+    }
+
+    if (getenv("TT_METAL_ENABLE_GATHERING")) {
+        this->enable_gathering = true;
     }
 
     const char *arc_debug_enabled_str = std::getenv("TT_METAL_ARC_DEBUG_BUFFER_SIZE");

--- a/tt_metal/llrt/rtoptions.hpp
+++ b/tt_metal/llrt/rtoptions.hpp
@@ -1,4 +1,4 @@
-// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+// SPDX-FileCopyrightText: © 2023 Tenstorrent AI ULC
 //
 // SPDX-License-Identifier: Apache-2.0
 

--- a/tt_metal/llrt/rtoptions.hpp
+++ b/tt_metal/llrt/rtoptions.hpp
@@ -15,7 +15,6 @@
 #include <map>
 #include <set>
 #include <string>
-#include <unordered_set>
 #include <vector>
 
 #include "core_coord.hpp"
@@ -157,6 +156,9 @@ class RunTimeOptions {
     // e.g. Store A followed by Load A will be unchanges but Store A followed by Load B may return B before A is written
     // This option will disable the relaxed ordering
     bool disable_relaxed_memory_ordering = false;
+
+    // Enable instruction gathering in Tensix core.
+    bool enable_gathering = false;
 
     // Buffer in DRAM to store various ARC processor samples. Feature not ready yet
     uint32_t arc_debug_buffer_size = 0;
@@ -358,6 +360,7 @@ public:
     inline bool get_hw_cache_invalidation_enabled() const { return this->enable_hw_cache_invalidation; }
 
     inline bool get_relaxed_memory_ordering_disabled() const { return this->disable_relaxed_memory_ordering; }
+    inline bool get_gathering_enabled() const { return this->enable_gathering; }
 
     tt_metal::DispatchCoreConfig get_dispatch_core_config() const;
 

